pkg/api/controller.go

```diff
@@ -1567,6 +1568,21 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 		return
 	}
 
+	if swag.BoolValue(body.SampleData) {
+		// add sample data, hooks, etc.
+		user, err := auth.GetUser(ctx)
+		if err != nil {
+			writeError(w, r, http.StatusUnauthorized, "missing user")
+			return
+		}
+
+		err = samplerepo.PopulateSampleRepo(ctx, newRepo, c.Catalog, c.PathProvider, c.BlockAdapter, user)
```

Comment:
@nopcoder 
I had a similar thought. Since the current requirements are mostly around templating in `README.md`, I'd prefer to use `remark` (which we already use anyway) and have the file be fully valid MD syntax, which will be rendered client-side instead of Go template syntax.

That being said, we can always use the templater package where needed. It should be easy enough to add a template expansion step in the `WalkDir` callback for `*.tt` files, specifically if/when we have this requirement.
WDYT?